### PR TITLE
Data Stores: Fix persistence data might be incorrect

### PR DESCRIPTION
**Pull request by arthur791004**
Related to #

## Proposed Changes

We use the `persistence` plugin to persist the store to the local storage on demand. However, if we register the plugin multiple times, the result will be incorrect and some of them will be overwritten with outdated data. Hence, this PR is proposing a change to wrap the registration in the single function to avoid the duplication.

## Testing Instructions

* Go to `/setup?siteSlug=<your_site>`
* Open the Devtool > Application > LocalStorage, and watch `WPCOM_7_DAYS_PERSISTENCE`
* When you back and forth between steps, the value of the `automattic/onboard` store should be updated.
  * For example, when you submit the selected goals, both goals, and intent will be updated to the selected values.
  * Note that the `verticalId` won't be updated since the vertical step doesn't set the value to the store. I tried to remove the `verticalId` from the store but looks like it's still used in the `storeProfiler` step

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/data-stores/src/launch/index.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export type { State };
 export type { LaunchStepType } from './types';
 export { STORE_KEY };
 
-use( plugins.persistence, persistOptions );
+registerPlugins();
```
Comment (taipeicoder) [inline packages/data-stores/src/launch/index.ts:13]: So many different ways this is implemented 🥲

**Review by taipeicoder — APPROVED** [review]
LGTM

File: packages/data-stores/src/launch/index.ts (hunk 1 above)
Comment (arthur791004) [reply packages/data-stores/src/launch/index.ts:13]: Yep...I also feel confused 🫠

**Review by okmttdhr — APPROVED** [review]
LGTM✅

Are there any other plugins we have implemented that might cause similar problems?

**Comment by arthur791004** [comment]
> Are there any other plugins we have implemented that might cause similar problems?

Looks like this is the only plugin we registered multiple times 🤔

**Pull request merged**